### ISSUE TITLE
Added CompoundVariant

### ISSUE THREAD

**Issue by kdfemi** [oktaysenkan/style-variants#2]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `style-variants@0.0.1` for the project I'm working on.

I added a compound variant props so that I can apply styles based on combined properties. An example is when I want to apply a certain style to an input when focused and is a certain variant. 
here is a sample code.

```
export const inputWrapperStyle = sv({
  base: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  variants: {
    variant: {
      clear: {
        borderBottomWidth: StyleSheet.hairlineWidth,
        borderColor: colors.border,
        paddingBottom: fonts.vr(12),
      },
      solid: {
        paddingHorizontal: fonts.hr(12),
        paddingVertical: fonts.vr(8),
        backgroundColor: colors.disabled,
        borderRadius: fonts.pt(50),
      },
    },
    noBorder: {
      true: {
        borderBottomWidth: 0,
      },
    },
    focused: {
      true: {},
    },
  },
  defaultVariants: {
    variant: 'clear',
    noBorder: true,
  },
  compoundVariants: [
    {
      style: {
        backgroundColor: colors.white,
        borderColor: colors.backgroundTint,
        borderWidth: StyleSheet.hairlineWidth,
        borderBottomWidth: StyleSheet.hairlineWidth,
      },
      variant: 'solid',
      noBorder: false,
      focused: true,
    },
  ],
});
```

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/style-variants/src/index.ts b/node_modules/style-variants/src/index.ts
index fe3fe1f..1317e10 100644
--- a/node_modules/style-variants/src/index.ts
+++ b/node_modules/style-variants/src/index.ts
@@ -1,8 +1,8 @@
-import type { ImageStyle, TextStyle, ViewStyle } from 'react-native';
+import type { ImageStyle, StyleProp, TextStyle, ViewStyle } from 'react-native';
 
 export type StringToBoolean<T> = T extends 'true' | 'false' ? boolean : T;
 
-export type Style = ViewStyle | TextStyle | ImageStyle;
+export type Style = StyleProp<ViewStyle | TextStyle | ImageStyle>;
 
 export type VariantProps<Component extends (...args: any) => any> =
   Parameters<Component>[0];
@@ -13,10 +13,17 @@ type VariantSchema<V extends VariantShape> = {
   [Variant in keyof V]?: StringToBoolean<keyof V[Variant]> | undefined;
 };
 
+type CompoundVariant<V extends VariantShape> = {
+  [Variant in keyof V]?: StringToBoolean<keyof V[Variant]> | undefined;
+} & {
+  style: Style;
+};
+
 export type Config<V extends VariantShape = VariantShape> = {
   base?: Style;
   variants?: V;
   defaultVariants?: VariantSchema<V>;
+  compoundVariants?: Array<CompoundVariant<V>>;
 };
 
 type Props<V> = V extends VariantShape
@@ -30,6 +37,7 @@ export const sv =
     base,
     defaultVariants,
     variants,
+    compoundVariants = [],
   }: Config<V>) =>
   (_options?: Props<V>) => {
     const styles: Style = {};
@@ -60,6 +68,17 @@ export const sv =
       }
     });
 
+    compoundVariants.forEach(compound => {
+      const {style, ...compoundVariantOptions} = compound;
+      if (
+        Object.entries(compoundVariantOptions).every(
+          ([key, value]) => options[key] === value,
+        )
+      ) {
+        Object.assign(styles, style);
+      }
+    });
+
     if (options.hasOwnProperty('style')) {
       Object.assign(styles, options.style);
     }
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by oktaysenkan** [comment]
Hi 🙋‍♂️

Thank you for contributing. Your input is greatly appreciated. I wanted to let you know that I will be adding this feature soon. Stay tuned for updates!

**Comment by franciscohanna92** [comment]
@oktaysenkan any updates on this?

**Comment by oktaysenkan** [comment]
It is available in v0.1.0

https://github.com/oktaysenkan/style-variants/releases/tag/v0.1.0
